resolwe_bio/processes/import_data/metadata.py

```diff
@@ -0,0 +1,153 @@
+"""Upload clinical metadata table."""
+from pathlib import Path
+
+import Orange
+
+from resolwe.process import (
+    FileField,
+    IntegerField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+def format_missing(missing):
+    """Format number of missing values as percentage."""
+    if missing:
+        return f"({missing:.1%} missing values)"
+    else:
+        return "(no missing values)"
+
+
+def get_features_description(data):
+    """Get formated features description."""
+    missing_features = format_missing(
+        data.has_missing_attribute() and data.get_nan_frequency_attribute()
+    )
+    return f"{len(data.domain.attributes)} {missing_features}"
+
+
+def get_target_description(data):
+    """Get formated target class description."""
+    missing_in_class = format_missing(
+        data.has_missing_class() and data.get_nan_frequency_attribute()
+    )
+
+    if data.domain.has_continuous_class:
+        target_description = f"Regression; numerical class {missing_in_class}"
+    elif data.domain.has_discrete_class:
+        target_description = (
+            "Classification; categorical class "
+            f"with {len(data.domain.class_var.values)} values {missing_in_class}"
+        )
+    elif data.domain.class_vars:
+        target_description = (
+            "Multi-target; "
+            f"{len(data.domain.class_vars)} target variables "
+            f"{missing_in_class}"
+        )
+    else:
+        target_description = "Not defined"
+
+    return target_description
+
+
+def change_suffix(path):
+    """Change suffix of a file to lowercase."""
+    new_path = path.with_suffix(path.suffix.lower())
+    path.replace(new_path)
+    return new_path
+
+
+class ImportMetadata(Process):
+    """Upload clinical metadata table in Orange format.
+
+    Orange can read files in native tab-delimited format, or can load
+    data from any of the major standard spreadsheet file types, like CSV
+    and Excel. Native format starts with the names of attributes with
+    prefixes that define atribute type (continuous, discrete, time,
+    string) and role (class, meta, ignore, instance weights). Prefixes
+    are separated from the attribute name with a hash sign (“#”).
+
+    Legacy format with three header rows is also supported. The first
+    row lists feature (column) names. The second header row gives the
+    attribute type and the third header line contains role information
+    to identify dependent features (class), irrelevant features (ignore)
+    or meta features (meta).
+
+    For more information see Orange
+    [documentation](https://orange-visual-programming.readthedocs.io/loading-your-data/index.html#header-with-attribute-type-information).
+
+    An example of native tab-delimited format can be downloaded
+    [here](http://file.biolab.si/datasets/sample-head.xlsx).
+
+    """
+
+    slug = "upload-clinical-metadata"
```

Comment:
`upload-orange-metadata` instead? This process focuses on importing some tabular data into orange, doesn't have to be clinical.